src/spec/junit_formatter.cr

```diff
@@ -104,7 +104,7 @@ module Spec
         HTML.escape(message, io)
         io << '"'
       end
-      if tag == :error
```

Comment:
Wait what? Was this a bug?

Reply:
Spec:

```crystal
require "spec"

it "errors" { raise "foo" }
```

Before:

```xml
<?xml version="1.0"?>
<testsuite tests="1" skipped="0" errors="1" failures="0" time="0.000237095" timestamp="2021-12-16T11:42:12Z" hostname="debian">
  <testcase file="/home/quinton/crystal/crystal/bug.cr" classname="bug" name="errors" time="7.5482000000000007e-05">
    <error message="foo">bug.cr:75:15 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;internal_run&#39;
src/spec/example.cr:33:16 in &#39;run&#39;
src/spec/context.cr:18:23 in &#39;internal_run&#39;
src/spec/context.cr:156:7 in &#39;run&#39;
src/spec/dsl.cr:220:7 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;run&#39;
src/crystal/main.cr:50:14 in &#39;exit&#39;
src/crystal/main.cr:45:5 in &#39;main&#39;
src/crystal/main.cr:124:3 in &#39;main&#39;
/lib/x86_64-linux-gnu/libc.so.6 in &#39;__libc_start_main&#39;
/home/quinton/.cache/crystal/crystal-run-spec.tmp in &#39;_start&#39;
???</error>
  </testcase>
</testsuite>
```

After:

```xml
<?xml version="1.0"?>
<testsuite tests="1" skipped="0" errors="1" failures="0" time="0.000288758" timestamp="2021-12-16T11:42:31Z" hostname="debian">
  <testcase file="/home/quinton/crystal/crystal/bug.cr" classname="bug" name="errors" time="7.3282e-5">
    <error message="foo" type="Exception">bug.cr:75:15 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;internal_run&#39;
src/spec/example.cr:33:16 in &#39;run&#39;
src/spec/context.cr:18:23 in &#39;internal_run&#39;
src/spec/context.cr:156:7 in &#39;run&#39;
src/spec/dsl.cr:220:7 in &#39;-&gt;&#39;
src/primitives.cr:266:3 in &#39;run&#39;
src/crystal/main.cr:50:14 in &#39;exit&#39;
src/crystal/main.cr:45:5 in &#39;main&#39;
src/crystal/main.cr:124:3 in &#39;main&#39;
/lib/x86_64-linux-gnu/libc.so.6 in &#39;__libc_start_main&#39;
/home/quinton/.cache/crystal/crystal-run-spec.tmp in &#39;_start&#39;
???</error>
  </testcase>
</testsuite>
```

I have updated a spec for this.